src/coreclr/vm/callingconvention.h

```diff
@@ -620,35 +632,35 @@ class ArgIteratorTemplate : public ARGITERATOR_BASE
 
         pLoc->m_fRequires64BitAlignment = m_fRequires64BitAlignment;
 
+        const unsigned byteArgSize = StackElemSize(GetArgSize());
         int cSlots = (GetArgSize() + 3) / 4;
-
         if (TransitionBlock::IsFloatArgumentRegisterOffset(argOffset))
         {
-            pLoc->m_idxFloatReg = (argOffset - TransitionBlock::GetOffsetOfFloatArgumentRegisters()) / 4;
-            pLoc->m_cFloatReg = cSlots;
+            pLoc->m_idxFloatReg = (argOffset - TransitionBlock::GetOffsetOfFloatArgumentRegisters()) / TARGET_POINTER_SIZE;
```

Comment:
Personally I would avoid the term `FLOAT_REGISTER_SIZE` on arm/arm64.  I would prefer `FLOAT_SIZE` or `SIZEOF_FLOAT`.

Reply:
Added `FLOAT_REGISTER_SIZE`, 

> I would prefer FLOAT_SIZE or SIZEOF_FLOAT.

would not it be confusing to have `#define FLOAT_SIZE 16` on arm64?

Reply:
I guess I didn't expect the value to be 16.  

- For Arm32, a float register i.e. `s0` would be 4 bytes, but it would be one of many in the `q0`/`v0` register.  
- For arm64 `s0` is the lower 4 bytes of the `v0` SIMD register which is 16 bytes.

`FLOAT_SIZE` is definitely, worse, but I still think `FLOAT_REGISTER_SIZE` might be ambiguous for ARM architectures. `VECTOR_REGISTER_SIZE` might be better for ARM/ARM64.

Reply:
The thing is that we use m_idxFloatReg, m_cFloatReg fields in the surrounding code, thus the FLOAT_REGISTER_SIZE matches those.